Engine/Module/ModuleLight.cpp

```diff
@@ -21,24 +28,104 @@ bool ModuleLight::CleanUp()
+		glUniform3fv(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".position").c_str()), 1, light->owner->transform.GetGlobalTranslation().ptr());
+		glUniform3fv(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".direction").c_str()), 1, light->owner->transform.GetFrontVector().ptr());
+		float cutOff_value = light->spot_light_parameters.cutoff;
+		glUniform1f(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".cutOff").c_str()), cutOff_value);
+		float outerCutOff_value = light->spot_light_parameters.outer_cutoff;
+		glUniform1f(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".outerCutOff").c_str()), outerCutOff_value);
+		float constant_value = light->spot_light_parameters.constant;
+		glUniform1f(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".constant").c_str()), constant_value);
+		float linear_value = light->spot_light_parameters.linear;
+		glUniform1f(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".linear").c_str()), linear_value);
+		float quadratic_value = light->spot_light_parameters.quadratic;
+		glUniform1f(glGetUniformLocation(program, std::string(spot_light_current_uniform_name + ".quadratic").c_str()), quadratic_value);
+
+		++current_number_spot_lights_rendered;
+		++i;
+	}
+
+	glUniform1i(glGetUniformLocation(program, "num_spot_lights"), current_number_spot_lights_rendered);
+}
+
+void ModuleLight::RenderPointLights(const float3& mesh_position, GLuint program)
+{
+	SortClosestLights(mesh_position, ComponentLight::LightType::POINT_LIGHT);
+	current_number_point_lights_rendered = 0;
+	int i = 0;
+	while (current_number_point_lights_rendered < MAX_POINT_LIGHTS_RENDERED && i < closest_lights.size())
+	{
+		ComponentLight* light = closest_lights[i].second;
+
+		float3 light_color_scaled = light->light_intensity * float3(light->light_color);
+		std::string point_light_current_uniform_name = "point_lights[" + std::to_string(current_number_point_lights_rendered) + "]";
+
+		glUniform3fv(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".color").c_str()), 1, light_color_scaled.ptr());
+		glUniform3fv(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".position").c_str()), 1, light->owner->transform.GetGlobalTranslation().ptr());
+		float constant_value = light->point_light_parameters.constant;
+		glUniform1f(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".constant").c_str()), constant_value);
+		float linear_value = light->point_light_parameters.linear;
+		glUniform1f(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".linear").c_str()), linear_value);
+		float quadratic_value = light->point_light_parameters.quadratic;
+		glUniform1f(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".quadratic").c_str()), quadratic_value);
```

Comment:
what if we use: 
glUniform1f(glGetUniformLocation(program, std::string(point_light_current_uniform_name + ".constant").c_str()), light->point_light_parameters.constant);
instead? we are declaring each frame a float variable when we can use directly the direction of the variable of the class.

Reply:
Yes, you are right!